src/battle_util.c

```diff
@@ -7664,3 +7685,20 @@ bool32 SetIllusionMon(struct Pokemon *mon, u32 battlerId)
 
     return FALSE;
 }
+
+bool8 ShouldGetStatBadgeBoost(u16 badgeFlag, u8 battlerId)
+{
+    if (B_BADGE_BOOST != GEN_3)
+        return FALSE;
+
+    if (gBattleTypeFlags & (BATTLE_TYPE_LINK | BATTLE_TYPE_EREADER_TRAINER | BATTLE_TYPE_x2000000 | BATTLE_TYPE_FRONTIER))
```

Comment:
change this to `else if` and delete the blank line before this (just for style, not for much other practical benefit)